crates/red_knot_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -222,16 +222,16 @@ impl SearchPaths {
 
         static_paths.push(stdlib_path);
 
-        let site_packages_paths = match site_packages_paths {
-            SitePackages::Derived { venv_path } => {
+        let site_packages_paths = match python_path {
+            PythonPath::SysPrefix(sys_prefix) => {
                 // TODO: We may want to warn here if the venv's python version is older
                 //  than the one resolved in the program settings because it indicates
                 //  that the `target-version` is incorrectly configured or that the
                 //  venv is out of date.
-                VirtualEnvironment::new(venv_path, system)
+                VirtualEnvironment::new(sys_prefix, system)
```

Comment:
I think we have to adjust the code here as well to support system python installations. I don't have a good sense for what's involved here. It might just be that we have to make the `pyvenv.cfg` step optional. I can tackle this with some guidance as part of this PR or create a follow up issue to add system python support.

Reply:
I think probably best to defer this -- it seems like a separate issue